src/plugins/intel_gpu/src/plugin/transformations/disable_f16_comp_for_periodic_funcs.cpp

```diff
@@ -0,0 +1,120 @@
+        {ov::op::util::ScatterNDBase::get_type_info_static().name, true},
+        {ov::op::util::ScatterElementsUpdateBase::get_type_info_static().name, true},
+        {ov::op::v8::Slice::get_type_info_static().name, true},
+        {ov::op::v1::Broadcast::get_type_info_static().name, true},
+        {ov::op::v0::Concat::get_type_info_static().name, true},
+        {ov::op::v1::Split::get_type_info_static().name, true},
+        {ov::op::v1::StridedSlice::get_type_info_static().name, true},
+        {ov::op::v0::Tile::get_type_info_static().name, true},
+        {ov::op::v16::Identity::get_type_info_static().name, true},
+        {ov::op::v1::Pad::get_type_info_static().name, true},
+        {ov::op::util::GatherNDBase::get_type_info_static().name, true},
+        {ov::op::util::GatherBase::get_type_info_static().name, true},
+        {ov::op::v15::Squeeze::get_type_info_static().name, true},
+        {ov::op::v0::Unsqueeze::get_type_info_static().name, true},
+        // Add more node types as needed
+    };
+
+    auto it = node_behavior_registry.find(node->get_type_info().name);
+    return it != node_behavior_registry.end() && it->second;
+}
+
+ov::intel_gpu::DisableFP16CompressionForPeriodicFuncs::DisableFP16CompressionForPeriodicFuncs()
+    : ov::pass::MatcherPass() {
+    set_name("DisableFP16CompressionForPeriodicFuncs");
+
+    auto cos = ov::pass::pattern::wrap_type<ov::op::v0::Cos>();
+    auto sin = ov::pass::pattern::wrap_type<ov::op::v0::Sin>();
+    auto sin_cos_func_pattern = std::make_shared<ov::pass::pattern::op::Or>(OutputVector{cos, sin});
+
+    ov::matcher_pass_callback callback = [&](ov::pass::pattern::Matcher& m) {
+        auto node = m.get_match_root();
+
+        if (node->get_input_element_type(0) != ov::element::f16
+            && node->get_input_element_type(0) != ov::element::f32) {
+            return false;
+        }
+
+        // Disable FP16 compression for the current node
+        ov::disable_fp16_compression(node);
+
+        // Traverse inputs to find the first node that modifies values
+        std::function<bool(const std::shared_ptr<ov::Node>&)> traverse_inputs = [&](const std::shared_ptr<ov::Node>& current_node) -> bool {
+            for (const auto& input : current_node->inputs()) {
+                auto next_node = input.get_source_output().get_node_shared_ptr();
+                if (auto constant = ov::as_type_ptr<ov::op::v0::Constant>(next_node)) {
+                    // If the next node is a constant, skip it
+                    continue;
+                }
+
+                if (auto parameter = ov::as_type_ptr<ov::op::v0::Parameter>(next_node)) {
+                    // If the next node is a parameter, skip it
+                    continue;
+                }
+
+                if (ov::fp16_compression_is_disabled(next_node)) {
+                    // If FP16 compression is already disabled, continue to the next input
+                    return true;
```

Comment:
why does it return true? I guess it should return false because it is not changing anything

Reply:
fixed it.